draft-ietf-quic-transport.md

```diff
@@ -1465,35 +1470,220 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+allow connections to survive changes to the client's address (client's IP
+address and/or port), such as those caused by a client migrating to a new
+network or a NAT rebinding changing the client's public address.
+
+
+### Path Validation and PMTU Verification
+
+PATH_CHALLENGE ({{frame-path-challenge}}) and PATH_RESPONSE
+({{frame-path-response}}) frames MAY be used for validating that an endpoint
+owns an address, or to verify the PMTU to/from an address.
+
+A server validates a client's address by sending a PATH_CHALLENGE frame
+containing a payload that is hard to guess to that address.  The server
+considers the address to be valid when a PATH_RESPONSE frame containing the same
+payload is received from the client.
+
+Similarly, a client verifies the PMTU of a network path from a new client
+address to the server by sending a PMTU-sized packet carrying a PATH_CHALLENGE
+frame. The client considers the PMTU verified when a PATH_RESPONSE frame
+containing the same payload sent by the server to the new client address is
+received.
+
+
+### Client Behavior {#migration-client}
+
+A client triggers connection migration to a new address by sending a
+PATH_CHALLENGE as a probe or by switching to sending all packets to the new
+address. The client SHOULD do PMTU verification before considering migration
+complete. The client also participates in an address validation that the server
+may initiate to confirm the client's ownership of its address (see
+{{migration-server}}).
+
+A client migrating to a new network might use a new connection ID for packets
+sent from the new address, see {{migration-linkability}} for further discussion.
+
+
+#### Verifying the PMTU
+
+When the client intends to use a new local address, it MAY establish server
+reachability from the new address, and it SHOULD verify the new path's MTU.  To
+achieve these ends, the client sends a packet containing a PATH_CHALLENGE frame
+from this new address before sending any other packets.  The PATH_CHALLENGE
+frame MAY be bundled with other frames, but this packet MUST be padded to
+exactly 1200 octets unless the client has a reasonable assurance that the PMTU
+is larger.  Sending a packet of this size confirms that the network path from
+the client to the server supports an MTU of this size.
+
+A client MAY send additional PATH_CHALLENGE frames to handle packet loss or to
+get more measurements on the new network path, subject to the congestion
+controller's limits {{QUIC-RECOVERY}}.
+
+The client SHOULD use fresh random data in every PATH_CHALLENGE frame so that it
+can perform more accurate measurements by associating the server's response with
+the causative PATH_CHALLENGE.
+
+When the client receives a PATH_RESPONSE from the server and verifies that the
+included data matches the data it sent in a previous PATH_CHALLENGE, the server
+is considered reachable from the new address and the PMTU of the new network
+path is considered verified.
```

Comment:
"the PMTU is considered verified" is a little too binary here.  You've mandated "exactly" 1200 bytes, which only verifies that 1200 is less than the PMTU.  The client would actually need to do a series of increasingly-large packets with PATH_CHALLENGEs before it actually knows the PMTU.  It might be worth integrating PATH_CHALLENGE into the existing PMTU section, referencing that, and simply saying that the client SHOULD re-probe PMTU whenever it becomes aware of a path change.